service/src/main/java/gov/nasa/pds/api/registry/controller/URIParameters.java

```diff
@@ -103,21 +102,15 @@ public URIParameters setKeywords(List<String> keywords)
 	}
 	public URIParameters setLimit(Integer limit)
 	{
-		/*
-		 * Note: Not too happy w/ having to put behavioral logic in a utility/container class, but
-		 * there are just way too many places where this information is necessary and rather than
-		 * duplicate it everywhere, this is the best place, for now, as it is the common object
-		 * shared across them all.
-		 */
-		if (limit != null) {
-			if(limit == 0) {
-				summaryOnly = true;
-				this.limit = Integer.valueOf(SUMMARY_SAMPLE_SIZE);
-			} else {
-				this.limit = limit;
-				summaryOnly = false;
-			}
+		if (limit == null) {return this;}
+
+		if (limit < 0) {
+			String errMsg = String.format("start index must be 0 or higher (got '%d'))", start);
+			throw new IllegalArgumentException(errMsg);
 		}
+
+		this.limit = limit;
+		this.returnSingularDatum = false;
```

Comment:
Somehow the property name is not meaningful to me. A bit too abstract maybe.

Reply:
This replaces `isSingular`.  The meaning is "this `URIParameters` is being used for an API route which returns a single element (ex. `/collections/{identifier}`) rather than a collection (ex. `/collections/{identifier}/products`).

Open to suggestions on this.

Reply:
@tloubrieu-jpl @alexdunnjpl ☝️

Reply:
@alexdunnjpl , What about something like `expectSingularResponse` ?

Reply:
I think it's slightly better, but I'm starting to feel like if it's worth worrying about, it's worth doing properly ([see the last bit of the comment in the next thread)[https://github.com/NASA-PDS/registry-api/pull/256#discussion_r1116109721]), so unless you have an objection I think that I should 
 - make it an enum type instead of a boolean
 - put an explanatory comment in the enum type so that anyone who wonders/cares what exactly it refers to can jump to the explanation with a minimum of effort

Reply:
Actually, how about `singletonResultExpected`?

Communicates that the expected singularness applies to the _result_ (of the query), and I think that "singleton" makes the connection a little clearer.